rtcp/src/transport_feedbacks/transport_layer_nack/mod.rs

```diff
@@ -27,40 +27,69 @@ pub struct NackPair {
 pub type RangeFn =
     Box<dyn (Fn(u16) -> Pin<Box<dyn Future<Output = bool> + Send + 'static>>) + Send + Sync>;
 
-impl NackPair {
-    /// PacketList returns a list of Nack'd packets that's referenced by a NackPair
-    pub fn packet_list(&self) -> Vec<u16> {
-        let mut out = vec![self.packet_id];
+pub struct NackIterator {
+    packet_id: u16,
+    bitfield: PacketBitmap,
+    index: Option<u16>,
+}
+
+impl Iterator for NackIterator {
+    type Item = u16;
 
-        let mut b = self.lost_packets;
-        let mut i = 0;
+    fn next(&mut self) -> Option<Self::Item> {
+        match self.index {
+            None => {
+                self.index = Some(0);
 
-        while b != 0 {
-            if (b & (1 << i)) != 0 {
-                b &= !(1 << i);
-                out.push(self.packet_id + i + 1);
+                Some(self.packet_id)
             }
-            i += 1;
+            Some(id) if self.bitfield != 0 => {
+                let mut i = id;
+
+                while self.bitfield != 0 {
+                    if (self.bitfield & (1 << i)) != 0 {
+                        self.bitfield &= !(1 << i);
+                        self.index = Some(i + 1);
+
+                        return Some(self.packet_id.wrapping_add(i + 1));
+                    }
+
+                    i += 1;
```

Comment:
This algo is a bit hard to read because what's technically one field have 3 different names: `self.index`, `id` and `i`. Would be good to be consistent.

Reply:
I'll fix